### PR TITLE
HDDS-3825. Show rule status of SCM safemode

### PR DESCRIPTION
**Pull request by maobaolong**
## What changes were proposed in this pull request?

Without this PR, we can't get the reason why scm still in the safemode, now, we can get the each rule status, and know the condition to exit safemode.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3825

## How was this patch tested?

```bash
ozone admin safemode status --verbose
SCM is in safe mode.
validated:false, DataNodeSafeModeRule, registeredDns 0 should >= requiredDns 1
validated:true, HealthyPipelineSafeModeRule, currentHealthyPipelineCount 0 should >= healthyPipelineThresholdCount 0
validated:true, ContainerSafeModeRule, currentContainerThreshold 1.0 should >= safeModeCutoff 0.99
validated:true, AtleastOneDatanodeReportedRule, currentReportedPipelineCount 0 should >= thresholdCount 0
```

### CODE REVIEW

**Comment by maobaolong** [comment]
@elek Please take a look, thank you.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/client/ScmClient.java [hunk 1]
```diff
@@ -210,6 +212,15 @@ Pipeline createReplicationPipeline(HddsProtos.ReplicationType type,
    */
   boolean inSafeMode() throws IOException;
 
+  /**
+   * Get the safe mode statuses of each rules.
```
Comment (dineshchitlangia) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/client/ScmClient.java:216]:
```suggestion
   * Get the safe mode status of all rules.
```

**Comment by maobaolong** [comment]
@dineshchitlangia Thanks for your review and the suggestion, i've commit all of your suggestion commit, thank for your help. PTAL.

**Comment by maobaolong** [comment]
@dineshchitlangia Thanks for your +1. 😃

**Comment by dineshchitlangia** [comment]
@maobaolong I will merge this. However, going forward could you pls also include any relevant tests with your changes as the code coverage is now being looked at more seriously in the community for the benefits it brings :)
For this patch, pls file another jira to add the tests.

**Comment by dineshchitlangia** [comment]
@maobaolong I filed https://issues.apache.org/jira/browse/HDDS-3849 and assigned to you Thanks.

**Comment by maobaolong** [comment]
@dineshchitlangia Thanks you for create the ticket for me, i'd love to do it with pleasure.

**Pull request merged**